File: .claude/skills/zero-trust-secret-orchestration/references/mjs/src/resolver.mjs
import { Strategy, StrategyConfigError } from "./strategy.mjs";

export async function createResolver({ strategies, env = process.env } = {}) {
  if (!Array.isArray(strategies) || strategies.length === 0) {
    throw new TypeError("createResolver requires a non-empty `strategies` array");
  }
  for (const s of strategies) {
    if (!(s instanceof Strategy)) {
      throw new TypeError(`chain entry ${s?.constructor?.name ?? typeof s} is not a Strategy`);
    }
  }

  const isDev = env.NODE_ENV === "development" || env.APP_ENV === "development";
  if (isDev) {
    const { SystemEnvStrategy } = await import("./strategies/system-env.mjs");
    if (strategies.some((s) => s instanceof SystemEnvStrategy)) {
      throw new StrategyConfigError(
        "SystemEnvStrategy is disabled in development. " +
          "Use KeychainBiometricStrategy locally to enforce the biometric gate.",
      );
    }
  }

  return {
    async resolve(key) {
      for (const s of strategies) {
        const v = await s.get(key);
        if (v !== null) return v;
      }
      return null;
    },
  };
}

File: .claude/skills/zero-trust-secret-orchestration/references/py/src/zts/resolver.py
from __future__ import annotations

import os
from dataclasses import dataclass
from typing import Sequence

from zts.strategy import Strategy, StrategyConfigError


@dataclass
class Resolver:
    strategies: Sequence[Strategy]

    async def resolve(self, key: str) -> str | None:
        for s in self.strategies:
            v = await s.get(key)
            if v is not None:
                return v
        return None


def create_resolver(
    strategies: Sequence[Strategy], env: dict | None = None
) -> Resolver:
    env = env if env is not None else os.environ
    if not strategies:
        raise TypeError("create_resolver requires a non-empty strategies sequence")
    for s in strategies:
        if not isinstance(s, Strategy):
            raise TypeError(f"chain entry {type(s).__name__} is not a Strategy")

    is_dev = env.get("APP_ENV") == "development" or env.get("NODE_ENV") == "development"
    if is_dev:
        from zts.strategies.system_env import SystemEnvStrategy

        if any(isinstance(s, SystemEnvStrategy) for s in strategies):
            raise StrategyConfigError(
                "SystemEnvStrategy is disabled in development. "
                "Use KeychainBiometricStrategy locally to enforce the biometric gate."
            )

    return Resolver(strategies=list(strategies))
